src/weapon/w_015.c

```diff
@@ -29,7 +29,46 @@ s32 func_ptr_80170004(Entity* self) {
     }
 }
 
-INCLUDE_ASM("weapon/nonmatchings/w_015", func_ptr_80170008);
+extern AnimationFrame D_6D000_8017A6C0[];
+extern s32 D_6D000_8017BFC8;
+
+void func_ptr_80170008(Entity* self) {
+    Entity* factory;
+
+    if (self->step == 0) {
+        self->animSet = 2;
+        self->unk4C = D_6D000_8017A6C0;
+        self->zPriority = PLAYER.zPriority - 4;
+        self->flags = FLAG_UNK_08000000 | FLAG_UNK_100000;
+        self->velocityY = FIX(-1);
+
+        if (!(D_6D000_8017BFC8 & 1)) {
+            factory =
+                g_api.CreateEntFactoryFromEntity(self, FACTORY(0xB00, 4), 0);
+            if (factory != NULL) {
+                if (g_HandId == 0) {
+                    factory->entityId = 0xEF;
+                } else {
+                    factory->entityId = 0xFF;
```

Comment:
Wow, this is an extremely weird way to do the factory, usually factories in the weapon overlay work g_HandId into the initial CreateEntFactoryFromEntity call. Really weird to see this, I wonder if this was put in before they had a standard way to do weapon factory stuff. Nice work!